lib/Basics/Result.h

```diff
@@ -18,89 +18,112 @@
 /// Copyright holder is ArangoDB GmbH, Cologne, Germany
 ///
 /// @author Dr. Frank Celler
+/// @author Dan Larkin-York
+/// @author Jan Christoph Uhde
 ////////////////////////////////////////////////////////////////////////////////
 
 #ifndef ARANGODB_BASICS_RESULT_H
 #define ARANGODB_BASICS_RESULT_H 1
 
 #include "Basics/Common.h"
+#include "Basics/error.h"
+#include <type_traits>
+#include <iostream>
```

Comment:
including iostream in a header file that is included about everywhere is not a good idea.
is it actually necessary at all? if yes, can we get away with `#include <iosfws>`?